cmd/kube-controller-manager/app/plugins.go

```diff
@@ -69,6 +74,23 @@ func ProbeRecyclableVolumePlugins(flags VolumeConfigFlags) []volume.VolumePlugin
 	return allPlugins
 }
 
+// getCreatableVolumePluginForCloud maps a cloud provider to a specific volume plugin.
+func getCreatableVolumePluginForCloud(cloud cloudprovider.Interface, server *CMServer) volume.CreatableVolumePlugin {
+	var creater volume.VolumePlugin
+	switch {
+	case cloud == nil && server.EnableHostPathProvisioner:
```

Comment:
Also, the hard coded index looks not right to me at all.  Do we need to have a special probe method that gives you back a single plugin?  It's completely arbitrary what someone could decide to return from this method and you're imposing a change to the contract by assimung `[0]` will be something that will be implement `CreatableVolumePlugin`.